src/modules/layout/Layout.jsx

```diff
@@ -71,11 +71,12 @@ const LayoutContent = () => {
                 componentsProps={{ button: { className: 'u-w-100 u-bdrs-6' } }}
                 label={t('upload.label')}
                 displayedFolder={displayedFolder}
+                disabled={isFolderReadOnly}
               />
               <AddMenuProvider
                 canCreateFolder={true}
-                canUpload={true}
-                disabled={false}
+                canUpload={!isFolderReadOnly}
```

Comment:
how about using only prop `disabled`, because the `UploadButton` is disabled as well if cannot upload to the displayed folder?